nixos/modules/services/networking/akkoma.nix

```diff
@@ -0,0 +1,522 @@
+
+                ```ShellSession
+                $ nix-shell -p nodejs --run 'npx web-push generate-vapid-keys'
+                ```
+              '';
+              type = types.submodule {
+                freeformType = format.type;
+                options = {
+                  ":vapid_details" = {
+                    subject = mkOption {
+                      type = types.str;
+                      example = "mailto:fediadmin@example.com";
+                      description = mdDoc "mailto URI for administrative contact.";
+                    };
+
+                    public_key = mkOption {
+                      type = types.str;
+                      description = mdDoc "base64-encoded public ECDH key.";
+                    };
+
+                    private_key = mkOption {
+                      type = secret;
+                      example = { _secret = "/run/keys/akkoma/vapid-private-key"; };
+                      description = mdDoc ''
+                        base64-encoded private ECDH key.
+
+                        The attribute `_secret` should point to a file containing the secret.
+                      '';
+                    };
+                  };
+                };
+              };
+            };
+
+            ":joken" = {
+              ":default_signer" = mkOption {
+                type = secret;
+                example = { _secret = "/run/keys/akkoma/jwt-signer"; };
+                description = mdDoc ''
+                  base64-encoded JWT signing secret.
+
+                  The attribute `_secret` should point to a file containing the secret.
+
+                  This secret can be generated as follows:
+
+                  ```ShellSession
+                  $ dd if=/dev/urandom bs=64 count=1 iflag=fullblock status=none | base64 -w 0
+                  ```
+                '';
+              };
+            };
+
+            ":tzdata" = {
+              ":data_dir" = mkOption {
+                type = types.str;
+                internal = true;
+                default = "${cfg.stateDir}/elixir_tzdata_data";
+                defaultText = literalExpression ''
+                  "''${config.services.akkoma.stateDir}/elixir_tzdata_data"
+                '';
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users = {
+      users."${cfg.user}" = {
+        description = "Akkoma user";
+        group = cfg.group;
+        isSystemUser = true;
+      };
+      groups."${cfg.group}" = { };
+    };
+
+    systemd.services.akkoma =
+      let
+        replaceSec =
+          let
+            replaceSec' = {}@args: v: with builtins;
+              if isSecret v then
+                hashString "sha256" v._secret
+              else if isAttrs v then
+                listToAttrs
+                  (lib.mapAttrsToList
+                    (key: val: { name = key; value = replaceSec' args val; })
+                    v)
```

Comment:
```suggestion
                lib.mapAttrs (_: val: replaceSec' args val)
```

Reply:
Thanks, that must be some relic from when I tried out various solutions to generating the config.